src/logger/Logger.cpp

```diff
@@ -0,0 +1,98 @@
+#include "kompute/logger/Logger.hpp"
+
+#if !KOMPUTE_OPT_LOG_LEVEL_DISABLED
+#if !KOMPUTE_OPT_USE_SPDLOG
```

Comment:
Would these functions still be used somewhere? Is it mainly on the tests?

Reply:
The Manager sets up the logger if required:
https://github.com/COM8/kompute/blob/b0f6535c4984ddf40e5ff4a7a2b11ad2215c1cab/src/Manager.cpp#L63-L65

So, yes this will happen also for the tests, in case this is what you mean here.

Reply:
Ok, just thinking from the conversation we can remove this from the core and move it to the tests as a separate PR, but should be ok for now

Reply:
Ah, yes this would be doable.